stock_warehouse_calendar_orderpoint/models/stock_warehouse_orderpoint.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from datetime import datetime, time
+
+from dateutil import relativedelta
+
+from odoo import api, fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = "stock.warehouse.orderpoint"
+
+    @api.depends(
+        "rule_ids",
+        "product_id.seller_ids",
+        "product_id.seller_ids.delay",
+        "warehouse_id.orderpoint_calendar_id",
+    )
+    def _compute_lead_days(self):
+        super()._compute_lead_days()
+        # Override to use the orderpoint calendar to compute the 'lead_days_date'
+        now = fields.Datetime.now()
+        for orderpoint in self.with_context(bypass_delay_description=True):
+            op_calendar = orderpoint.warehouse_id.orderpoint_calendar_id
+            if not op_calendar:
+                continue
+            if not orderpoint.product_id or not orderpoint.location_id:
+                orderpoint.lead_days_date = False
+                continue
+            lead_days = orderpoint._get_lead_days()
+            # Get the next planned date to execute this orderpoint
+            start_date = op_calendar.plan_days(1, now)
+            lead_days_date = start_date + relativedelta.relativedelta(days=lead_days)
+            # Postpone to the next available workday if needed
+            if (
+                orderpoint.warehouse_id.orderpoint_on_workday
+                and orderpoint.warehouse_id.calendar_id
+            ):
+                lead_days_date = datetime.combine(lead_days_date.date(), time.min)
```

Comment:
If lead_days_date is `2022-06-09 23:00:00` and wh TZ is `Europe/Paris`, then the datetime should be `2022-06-10 01:00:00`.
Also, `combine(datetime, time.min)` is enough.

Reply:
For the second remark you mean this?
```python
lead_days_date = datetime.combine(lead_days_date, time.min)
```
:+1: 
And I guess there is no need to play with the TZ if I do this.

Reply:
No, because lead_days_date isn't TZ aware. If you apply time.min on it, you might get the wrong day:
- E.G -> 2022-06-02 00:30:00 would become 2022-06-02 00:00:00
But if it is supposed to be UTC+2, we should have 2022-06-01 22:30:00 which would become 2022-06-01 00:00:00

Reply:
I would say that the rule of thumb is to make a datetime TZ aware before:
- `datetime.date()`
- `combine(datetime, time.min/max)`
- …

And to convert it to a naive datetime just before storing the datetime in the DB.

Reply:
Everything has been updated and also tested with a delay as float (with https://github.com/OCA/purchase-workflow/pull/1484) to ensure upcoming compatibility. Thanks for your help.